pkg/management/component.go

```diff
@@ -2559,3 +2564,78 @@ func (c *component) GetIdentityProviders(ctx context.Context, realmName string)
 
 	return apiIdps, nil
 }
+
+func (c *component) GetThemeConfiguration(ctx context.Context, realmName string) (api.ThemeConfiguration, error) {
+	// get the realm config from Keycloak
+	backofficeConfig, err := c.configDBModule.GetAdminConfiguration(ctx, realmName)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return api.ThemeConfiguration{}, err
+	}
+
+	if backofficeConfig.BoTheme == nil {
+		c.logger.Warn(ctx, "msg", "No theme configured for realm")
+		return api.ThemeConfiguration{}, errorhandler.CreateNotFoundError("theme")
+	}
+
+	config, err := c.configDBModule.GetThemeConfiguration(ctx, *backofficeConfig.BoTheme)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return api.ThemeConfiguration{}, err
+	}
+
+	// Convert the configuration to API representation
+	configAPI := api.ConvertToThemeConfiguration(config)
+
+	return configAPI, nil
+}
+
+func (c *component) UpdateThemeConfiguration(ctx context.Context, realmName string, themeConfig api.UpdatableThemeConfiguration) error {
+
+	// Convert the API representation to the database representation
+	config := configuration.ThemeConfiguration{
+		ThemeName: themeConfig.ThemeName,
+		Settings:  themeConfig.Settings,
+		Logo:      themeConfig.Logo,
+		Favicon:   themeConfig.Favicon,
+	}
+
+	config.Translations = themeConfig.Translations
+
+	// Update the theme configuration in the database
+	err := c.configDBModule.UpdateThemeConfiguration(ctx, config)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (c *component) GetThemeTranslation(ctx context.Context, realmName string, language string) (any, error) {
+	// get the realm config from Keycloak
+	backofficeConfig, err := c.configDBModule.GetAdminConfiguration(ctx, realmName)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return "", err
+	}
+
+	if backofficeConfig.BoTheme == nil {
+		c.logger.Warn(ctx, "msg", "No theme configured for realm")
+		return "", errorhandler.CreateNotFoundError("theme")
+	}
+
+	translationStr, err := c.configDBModule.GetThemeTranslation(ctx, *backofficeConfig.BoTheme, language)
```

Comment:
Should we force language to be lowercase or uppercase to avoid inconsistencies in database?